Products/ZenModel/Device.py

```diff
@@ -2235,4 +2236,22 @@ def getMacAddressCache(self):
     def getMacAddresses(self):
         return list(self.macaddresses or [])
 
+    security.declareProtected(access_contents_information, 'propertyItems')
+    def propertyItems(self):
+        """Return a list of (id, property) tuples. If property id related
+           to password, it will mask property value with '***'
+        """
+        result = []
+        for prop in self._properties:
+            prop_id = prop['id']
+            if prop['type'] == 'password':
+                prop_value = "***"
```

Comment:
I'd recommend searching for the callers of `propertyItems`. It appears to be used in the ZenCollector collector framework to get properties needed for collection. So returning asterisks instead of the proper password seems like it would break all sorts of things.